### PR TITLE
bugfix: Change Java home on startup if broken

### PR DESCRIPTION
**Pull request by tgodzik**
@kpodsiad this should also fix the issue on windows, what do you think?

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/metals/BloopServers.scala [hunk 1]
```diff
@@ -402,29 +402,46 @@ final class BloopServers(
...
+        maybeLoadBloopGlobalJsonFile(bloopPath) match {
+          case (Some(javaHome), opts) =>
+            Try {
+              val homePath = AbsolutePath(Paths.get(javaHome))
+              // fix java home in case it changed
+              if (!homePath.exists)
+                writeJVMPropertiesToBloopGlobalJsonFile(
+                  opts,
+                  metalsJavaHome
+                )
```
Comment (kpodsiad) [inline metals/src/main/scala/scala/meta/internal/metals/BloopServers.scala:430]: I would add here two `scribe.info`s here. First one just after if - `detected non existing java path in <path-to-bloop-config> file. The second one after overwriting it `overwrited it with <metalsJava>`

**Review by kpodsiad — APPROVED** [review]
Both idea and code looks code. Even though I don't have time to check this myself if it works just as it looks like I'll trust it works. LGTM

**Review by ckipp01 — APPROVED** [review]

File: metals/src/main/scala/scala/meta/internal/metals/BloopServers.scala (hunk 1 above)
Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/metals/BloopServers.scala]: Added!

Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/metals/BloopServers.scala]: I also added info about which Java home and options is used currently for Bloop and where it is defined, so the users can potentially debug things easier themselves.

And also the first set up should be now logged.

**Pull request merged**